frontend/src/components/Forms/RegisterForm.js

```diff
@@ -0,0 +1,257 @@
+    { name: tc("passwordRule2"), re: /\d/ },
+    {
+      name: tc("passwordRule3"),
+      re: /[-._!`'#%&,:;<>=@{}~$()*+/?[\]^|]/,
+    },
+    { name: tc("passwordRule4"), re: /[A-Z]/ },
+    { name: tc("passwordRule5"), re: /^\S*$/ },
+    { name: tc("passwordRule6"), re: /(?=.{8,})/ },
+  ];
+
+  const onChangePassword = ({ target }) => {
+    const criteria = checkBoxOptions
+      .map((x) => {
+        const available = x.re.test(target.value);
+        return available ? x.name : false;
+      })
+      .filter((x) => x);
+    setCheckedList(criteria);
+  };
+
+  const onFinish = (values) => {
+    console.info("values", values);
+    router.push("/login");
+  };
+
+  return (
+    <Form onFinish={onFinish}>
+      <Form.Item
+        name="fullname"
+        rules={[
+          {
+            required: true,
+            message: tc("fullNameRequired"),
+          },
+        ]}
+      >
+        <Input
+          placeholder={t("fullName")}
+          prefix={<UserCircle />}
+          variant="borderless"
+        />
+      </Form.Item>
+      <Form.Item
+        name="gender"
+        rules={[
+          {
+            required: true,
+            message: tc("genderRequired"),
+          },
+        ]}
+      >
+        <Select
+          placeholder={t("gender")}
+          options={genderOptions}
+          variant="borderless"
+        />
+      </Form.Item>
+      <Form.Item
+        name="country"
+        rules={[
+          {
+            required: true,
+            message: tc("countryRequired"),
+          },
+        ]}
+      >
+        <Select
+          placeholder={t("country")}
+          options={countryOptions}
+          fieldNames={{ label: "name", value: "alpha-2" }}
+          optionFilterProp="name"
+          variant="borderless"
+          showSearch
+          allowClear
+        />
+      </Form.Item>
+      <Form.Item
+        name="purpose_account"
+        rules={[
+          {
+            required: true,
+            message: tc("purposeAccountRequired"),
+          },
+        ]}
+      >
+        <Select
+          placeholder={t("purposeAccount")}
+          options={purposeOptions}
+          variant="borderless"
+        />
+      </Form.Item>
+      <Form.Item
+        name="email"
+        rules={[
+          {
+            required: true,
+            type: "email",
+            message: tc("emailRequired"),
+          },
+        ]}
+      >
+        <Input
+          placeholder={t("email")}
+          type="email"
+          prefix={<Envelope />}
+          variant="borderless"
+        />
+      </Form.Item>
+      <Form.Item
+        name="password"
+        rules={[
+          {
+            required: true,
+          },
+          () => ({
+            validator() {
+              if (checkedList.length === 6) {
+                return Promise.resolve();
+              }
+              return Promise.reject(new Error(tc("passwordCriteriaError")));
```

Comment:
Should inform the error in detail (add info icon) after "passwordCriteriaError" with tooltip: list of criteria.